### PR TITLE
Complete ReadMe \ Installation steps for Google Cloud Build Plugin

### LINKED ISSUE #19514: 🐛 Bug Report: Cloudbuild doesn't work
### 📜 Description

Hi team, I've been trying to configure the Cloudbuild plugin, but it seems to be quite outdated, as the full integration is not working as it should.
In the component-info.yaml, it doesn't seem to recognize the annotation google.com/cloudbuild-project-slug, and when I start Backstage, the message 'No CI/CD available for this entity' appears.

Following the Roadie documentation (https://roadie.io/backstage/plugins/google-cloud-build/), the error seems to be in the configuration of EntityPage.tsx.

### 👍 Expected behavior

Cloudbuild should work properly

### 👎 Actual Behavior with Screenshots

Doesn't recognize de google.com/cloudbuild-project-slug.
image

### 👟 Reproduction steps

Just follow the Roadie documentation https://roadie.io/backstage/plugins/google-cloud-build/

### 📃 Provide the context for the Bug.

I need to show a PoC of Backstage working with Cloudbuild 

### 🖥️ Your Environment

OS:   Darwin 22.6.0 - darwin/x64
node: v18.0.0
yarn: 1.22.19
cli:  0.22.11 (installed)
backstage:  1.17.2

Dependencies:
  @backstage/app-defaults                          1.4.2
  @backstage/backend-app-api                       0.5.1
  @backstage/backend-common                        0.15.2, 0.19.3
  @backstage/backend-dev-utils                     0.1.1
  @backstage/backend-openapi-utils                 0.0.3
  @backstage/backend-plugin-api                    0.6.1
  @backstage/backend-tasks                         0.5.6
  @backstage/catalog-client                        1.4.3
  @backstage/catalog-model                         1.4.1
  @backstage/cli-common                            0.1.12
  @backstage/cli-node                              0.1.3
  @backstage/cli                                   0.22.11
  @backstage/config-loader                         1.4.0
  @backstage/config                                1.0.8
  @backstage/core-app-api                          1.9.1
  @backstage/core-components                       0.11.2, 0.13.4
  @backstage/core-plugin-api                       1.5.3
  @backstage/errors                                1.2.1
  @backstage/eslint-plugin                         0.1.3
  @backstage/integration-aws-node                  0.1.5
  @backstage/integration-react                     1.1.17
  @backstage/integration                           1.6.1
  @backstage/plugin-api-docs                       0.9.9
  @backstage/plugin-app-backend                    0.3.49
  @backstage/plugin-app-node                       0.1.1
  @backstage/plugin-auth-backend                   0.18.7
  @backstage/plugin-auth-node                      0.2.18
  @backstage/plugin-catalog-backend                1.12.1
  @backstage/plugin-catalog-common                 1.0.15
  @backstage/plugin-catalog-graph                  0.2.34
  @backstage/plugin-catalog-import                 0.9.12
  @backstage/plugin-catalog-node                   1.4.2
  @backstage/plugin-catalog-react                  1.8.2
  @backstage/plugin-catalog                        1.12.3
  @backstage/plugin-cloudbuild                     0.3.22
  @backstage/plugin-events-node                    0.2.10
  @backstage/plugin-gcp-projects                   0.3.40
  @backstage/plugin-github-actions                 0.6.3
  @backstage/plugin-org                            0.6.12
  @backstage/plugin-permission-common              0.7.7
  @backstage/plugin-permission-node                0.7.12
  @backstage/plugin-permission-react               0.4.14
  @backstage/plugin-proxy-backend                  0.3.1
  @backstage/plugin-scaffolder-backend             1.16.2
  @backstage/plugin-scaffolder-common              1.4.0
  @backstage/plugin-scaffolder-node                0.2.1
  @backstage/plugin-scaffolder-react               1.5.3
  @backstage/plugin-scaffolder                     1.14.3
  @backstage/plugin-search-backend-module-catalog  0.1.5
  @backstage/plugin-search-backend-module-pg       0.5.10
  @backstage/plugin-search-backend-module-techdocs 0.1.5
  @backstage/plugin-search-backend-node            1.2.5
  @backstage/plugin-search-backend                 1.4.1
  @backstage/plugin-search-common                  1.2.5
  @backstage/plugin-search-react                   1.6.4
  @backstage/plugin-search                         1.3.5
  @backstage/plugin-tech-radar                     0.6.7
  @backstage/plugin-techdocs-backend               1.6.6
  @backstage/plugin-techdocs-module-addons-contrib 1.0.17
  @backstage/plugin-techdocs-node                  1.7.5
  @backstage/plugin-techdocs-react                 1.1.9
  @backstage/plugin-techdocs                       1.6.7
  @backstage/plugin-user-settings                  0.7.7
  @backstage/release-manifests                     0.0.9
  @backstage/test-utils                            1.4.2
  @backstage/theme                                 0.2.19, 0.4.1
  @backstage/types                                 1.1.0
  @backstage/version-bridge                        1.0.4

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by PurseChicken**
## Hey, I just made a Pull Request!

This PR fixes #19514

I modified the existing readme.md file to include all the information required to successfully install and use the Google Cloud Build plugin.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: thanks for this!

Think this is a good step in the right direction for sure.

I wonder though if we can take some inspiration from the `catalog` plugin and show `diff` patches for where code should be inserted to keep them easier to follow?

Thoughts?

**Comment by PurseChicken** [comment]
> Hey 👋 thanks for this!
> 
> Think this is a good step in the right direction for sure.
> 
> I wonder though if we can take some inspiration from the `catalog` plugin and show `diff` patches for where code should be inserted to keep them easier to follow?
> 
> Thoughts?

Sorry for the delay here.

I went ahead and reviewed the catalog ReadMe and implemented the diff's into this ReadMe via the latest commit. Let me know if that's in alignment with what you were thinking.

Thanks!

File: plugins/cloudbuild/README.md [hunk 1]
```diff
@@ -1,13 +1,81 @@
...
+```diff
+// packages/app/src/components/catalog/EntityPage.tsx
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+
+import { ReportIssue } from '@backstage/plugin-techdocs-module-addons-contrib';
+
++import { EntityCloudbuildContent, isCloudbuildAvailable } from '@backstage/plugin-cloudbuild';
+```
+
+#### In your 'cicdContent' const, add the following switch case
```
Comment (vinzscam) [inline plugins/cloudbuild/README.md:33]:
```suggestion
#### In your `cicdContent` const, add the following switch case
```

**Review by benjdlambert — APPROVED** [review]
Thanks! :tada:

**Comment by benjdlambert** [comment]
@PurseChicken one thing, we will need to you to sign the commits using DCO. You can see how to do that in the comment: https://github.com/backstage/backstage/pull/20812#issuecomment-1780097621

**Comment by PurseChicken** [comment]
> @PurseChicken one thing, we will need to you to sign the commits using DCO. You can see how to do that in the comment: [#20812 (comment)](https://github.com/backstage/backstage/pull/20812#issuecomment-1780097621)

I folled the documentation and it seems because I synced my fork its trying to pull in a bunch of other commits due to the rebase. Should I just redo this PR?

**Comment by freben** [comment]
@PurseChicken maybe it's easiest at this point - there's _always_ some git magic you can pull to get out of a hole like this, but sometimes it's just easier to just start over quickly

**Comment by PurseChicken** [comment]
I am closing this PR and will be re-submitting.

**Comment by PurseChicken** [comment]
New PR is at https://github.com/backstage/backstage/pull/21069

**Pull request closed without merging**